### ISSUE TITLE
C++ Fix ups

### ISSUE THREAD

**Issue by BenjamenMeyer** [vegastrike/Vega-Strike-Engine-Source#290]
There's a number of things we should clean up to make our lives easier. This issue is going to track several of those as we come across them:

- [ ] good refactor would be to take things like vector< StarSystem*> and make them types:
```
typedef vector<StarSystem*> StarSystemList;
```
and then use that new type everywhere instead of the more expansive form. This will also help eliminate bugs, but also make typing easier. It also means we only get one definition of it which should also help with compile times.
  - [ ] These should be placed in the header file as close to the definition of the original item as possible.

- [ ] All data-only objects should be structures.
- [ ] All data+method objects should be classes.
- [ ] All definitions should be in header files as much as possible.
  - It should be the exception, not the rule, that things get defined in implementation files. (Thanks @royfalk for taking care of a lot of this in the refactors thus far.)

**Comment by stephengtuggy** [comment]
At some point, I'd like to get the entire project converted over to using smart pointers, as much as possible. (#8 ) So at that point, `StarSystemList` might be redefined as `vector<boost::shared_ptr<StarSystem>>` or something like that.

Does that sound reasonable?

**Comment by BenjamenMeyer** [comment]
@stephengtuggy yeah, that'd be reasonable this change would enable that to be more seamless across the codebase so we'd probably want to do this first.

**Comment by BenjamenMeyer** [comment]
@stephengtuggy updated the example to also typedef the pointer; for the change in #8 once this (#290) is implemented then we'd simply update the one typedef and everything else would get updated automatically. Little things like this can make maintenance a lot easier.

**Comment by stephengtuggy** [comment]
@BenjamenMeyer sounds good.

**Comment by BenjamenMeyer** [comment]
We've probably been doing this with the various refactors.
@royfalk should we close this?
